vmdb/app/models/ems_refresh/save_inventory.rb

```diff
@@ -176,7 +176,7 @@ def save_guest_devices_inventory(hardware, hashes)
       end
     end
 
-    deletes = hardware.guest_devices.find(:all, :conditions => {:device_type => ['ethernet', 'storage']}).dup
+    deletes = hardware.guest_devices.where(:device_type => ["ethernet", "storage"]).to_a.dup
```

Comment:
It appears to return the same array for the same AR::Relation, so, maybe the dup is still needed, cc @tenderlove @matthewd 

```ruby
irb(main):023:0> pf = MiqProductFeature.limit(2).select(:id)
=> [#<MiqProductFeature id: 1>, #<MiqProductFeature id: 2>]
irb(main):024:0> pf.to_a.object_id
=> 70215016980300
irb(main):025:0> pf.to_a.object_id
=> 70215016980300
```